File: AlgorithmForCompetition/Greedy/AdventurerGuild.js
// 풀이 시간 30 => 12:11 - 12:41

const N = 5;
const guild = [2,3,1,2,2];
// const guild = [8,1,1,1,1,1,1,1];
// const guild = [2,3,3];
//-----------------------------[input]

let group = 0;
let count = 0;
guild.sort((a,b) => a-b);

for(let i of guild) {
  count += 1;
  if(count >= i) {
    group++;
    count = 0;
  }
}

console.log(group);

File: RestoreAlgorithm/Greedy/AdventurerGuild.cpp
// ProblemNumber || ProblemName : 모험가 길드

#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

int main()
{
  int N;
  int cnt = 0;
  vector<int> arr;
  cin >> N;

  for (int i = 0; i < N; i++)
  {
    int temp;
    cin >> temp;
    arr.push_back(temp);
  }

  sort(arr.begin(), arr.end());

  int guild = 0;

  for (int i = 0; i < arr.size(); i++)
  {
    guild++;
    if (arr[i] <= guild)
    {
      cnt++;
      guild = 0;
    }
  }

  cout << cnt << endl;
}
